src/audio_unit/looper.rs

```diff
@@ -88,24 +110,72 @@ impl Looper {
                 }
             }
             Playing { position, total } => {
+                self.process_samples_playing(position, total, input, output);
+            }
+            PlayingAwaitingOverdub { position, total } => {
+                self.process_samples_playing(position, total, input, output);
+            }
+            Overdubbing { position, total } => {
                 let next_position = position + output.len();
 
                 if next_position <= total {
                     output.copy_from_slice(&self.buffer[position..next_position]);
+
+                    for sample_number in 0..input.len() {
+                        self.buffer[position + sample_number] =
+                            self.buffer[position + sample_number] + input[sample_number];
+                    }
+
                     let position = if next_position == total {
                         0
                     } else {
                         next_position
                     };
 
-                    self.state = Playing { position, total }
+                    self.state = match next_position == total {
```

Comment:
Is this idiomatic to use `match` where I might otherwise reach for a ternary?

Reply:
I'd probably use an if-expression instead to avoid having to write out the cases.

Reply:
Ah I didn't know you could use `if`/`else` as an expression in Rust, updated